Realm/RLMArray.mm

```diff
@@ -299,6 +299,25 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
```

Comment:
I'd typically write this using an early-return to avoid nesting the entirety of the method, and to make it clear that we never call through to `super` when we have a backing array. I'm not sure if we have a general policy on that though.